### PR TITLE
ST-205 refactor Payment.ts class

### PR DESCRIPTION
**Pull request by pgs-gmartowski**

### CODE REVIEW

File: src/core/shared/Payment.ts [hunk 1]
```diff
@@ -59,7 +85,15 @@ export default class Payment {
       merchantData,
```
Comment (pgs-gmartowski) [inline src/core/shared/Payment.ts]: @james-simpson-ST is `termurl: 'https://termurl.com'` still needed ?

File: src/core/integrations/ApplePay.ts [hunk 2]
```diff
@@ -134,7 +135,7 @@ export class ApplePay {
    * If yes, returns ApplePaySession object, if not returns undefined.
    */
   public ifApplePayIsAvailable() {
-    return ApplePaySession ? true : false;
+    return !!ApplePaySession;
```
Comment (pgs-gmartowski) [inline src/core/integrations/ApplePay.ts]: it was tslint hint - it really looks less obscure than ternary above

File: src/core/shared/Payment.ts (hunk 1 above)
Comment (james-simpson-ST) [reply src/core/shared/Payment.ts]: Last time I checked this had not been done by the gateway team :( I will re-raise it with them

File: src/core/integrations/ApplePay.ts (hunk 2 above)
Comment (james-simpson-ST) [reply src/core/integrations/ApplePay.ts]: Makes sense to me

**Pull request merged**